File: src/main/java/web/controlevacinacao/repository/ClientRepository.java
package web.controlevacinacao.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import web.controlevacinacao.model.fidex_model.Status;
import web.controlevacinacao.model.fidex_model.Client;
import web.controlevacinacao.repository.queries.client.ClientQueries;

public interface ClientRepository extends JpaRepository<Client, Long>, ClientQueries {
    List<Client> findByStatus(Status status);
}

File: src/main/kotlin/web/fidex/repository/ClientRepository.kt
package web.fidex.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import web.fidex.model.fidex_model.Client
import web.fidex.model.fidex_model.Status
import web.fidex.repository.queries.client.ClientQueries

interface ClientRepository : JpaRepository<Client, Long>, ClientQueries {
    fun findByStatus(status: Status): List<Client>
    fun findByCreatedBy(userId: String, pageable: Pageable): Page<Client>
}
